plenum/server/node.py

```diff
@@ -386,7 +387,8 @@ def __init__(self,
             (CatchupReq, self.ledgerManager.processCatchupReq),
             (CatchupRep, self.ledgerManager.processCatchupRep),
             (CurrentState, self.process_current_state_message),
-            (ObservedData, self.send_to_observer)
+            (ObservedData, self.send_to_observer),
+            (BackupInstanceFaulty, self.process_backup_instance_faulty_msg)
```

Comment:
Maybe we can put all logic related to backup instance removing to a separate class (BackupInstanceFaultyProcessor)?
It can contains the following methods:
- process_backup_instance_faulty_msg
- on_backup_degradation
- restore
- __send_backup_instance_faulty
- __remove_replica